common/coconut-interface/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+        let verification_key = get_aggregated_verification_key(validator_urls).unwrap();
+        let params = Parameters::new(self.n_params).unwrap();
+        coconut_rs::verify_credential(
+            &params,
+            &verification_key,
+            &self.theta,
+            &self.public_attributes(),
+        )
+    }
+}
+
+#[derive(Getters, CopyGetters)]
+pub struct VerifyCredentialBody {
+    #[getset(get = "pub")]
+    n_params: u32,
+    #[getset(get = "pub")]
+    theta: Theta,
+    public_attributes: Vec<String>,
+}
+
+impl VerifyCredentialBody {
+    pub fn new(
+        n_params: u32,
+        theta: &Theta,
+        public_attributes: &[Attribute],
+    ) -> VerifyCredentialBody {
+        VerifyCredentialBody {
+            n_params,
+            theta: theta.clone(),
+            public_attributes: public_attributes
+                .iter()
+                .map(|attr| attr.to_bs58())
+                .collect(),
+        }
+    }
+
+    pub fn public_attributes(&self) -> Vec<Attribute> {
+        self.public_attributes
+            .iter()
+            .map(|x| Attribute::try_from_bs58(x).unwrap())
+            .collect()
+    }
+}
+//  All strings are base58 encoded representations of structs
+#[derive(Serialize, Deserialize, Debug, Getters, CopyGetters)]
+pub struct BlindSignRequestBody {
+    #[getset(get = "pub")]
+    blind_sign_request: BlindSignRequest,
+    #[getset(get = "pub")]
+    public_key: PublicKey,
+    public_attributes: Vec<String>,
+    #[getset(get = "pub")]
+    total_params: u32,
+}
+
+impl BlindSignRequestBody {
+    pub fn new(
+        blind_sign_request: &BlindSignRequest,
+        public_key: &PublicKey,
+        public_attributes: &[Attribute],
+        total_params: u32,
+    ) -> BlindSignRequestBody {
+        BlindSignRequestBody {
+            blind_sign_request: blind_sign_request.clone(),
+            public_key: public_key.clone(),
+            public_attributes: public_attributes
+                .iter()
+                .map(|attr| attr.to_bs58())
+                .collect(),
+            total_params,
+        }
+    }
+
+    pub fn public_attributes(&self) -> Vec<Attribute> {
+        self.public_attributes
+            .iter()
+            .map(|x| Attribute::try_from_bs58(x).unwrap())
+            .collect()
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct BlindedSignatureResponse {
+    pub blinded_signature: BlindedSignature,
+}
+
+impl BlindedSignatureResponse {
+    pub fn new(blinded_signature: BlindedSignature) -> BlindedSignatureResponse {
+        BlindedSignatureResponse { blinded_signature }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct VerificationKeyResponse {
+    pub key: VerificationKey,
+}
+
+impl VerificationKeyResponse {
+    pub fn new(key: VerificationKey) -> VerificationKeyResponse {
+        VerificationKeyResponse { key }
+    }
+}
+
+fn get_verification_key(url: &str) -> Result<VerificationKey, String> {
```

Comment:
Considering all of our other code is working nicely in async context, shouldn't we rather use some async library for http calls? For example `reqwest` that we already use elsewhere?